internal/consensus/reactor.go

```diff
@@ -286,9 +284,8 @@ func (conR *Reactor) Receive(e p2p.Envelope) {
 		case *HasProposalBlockPartMessage:
 			ps.ApplyHasProposalBlockPartMessage(msg)
 		case *VoteSetMaj23Message:
-			conR.rsMtx.RLock()
-			height, votes := conR.rs.Height, conR.rs.Votes
-			conR.rsMtx.RUnlock()
+			rs := conR.getRoundState()
```

Comment:
We are operating in a slightly possibly outdated version of the round state here.

At the same time, this message is not very common (it is protected by a 2s sleep, see `queryMaj23Routine` method).

We probably here could get a copy of the consensus state instead of relying on the local copy we keep in the reactor.

Reply:
I am pretty sure that the profiling results would not change if we use here the latest version of the consensus state, instead of the (possibly stale) local copy.

Reply:
Done